### PR TITLE
Typeahead

### PR DESCRIPTION
**Pull request by danhenderson**
Typeahed/InputSearch PR take 2.

This time a lot closer to our other input components, and playing very nicely with withForm.

I tried to keep the component flexible e.g. you still have to BYO filter function, which allows us to perform everything from simple synchronous filtering on a known dataset, or asynchronous requests to an API to get filtered results.

typeahead-2 0

### CODE REVIEW

File: source/components/input-search/styles.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    },
+
+    selected: {
+      position: 'absolute',
+      display: 'flex',
+      alignItems: 'stretch',
+      left: rhythm(0.175),
+      top: rhythm(0.175),
+      bottom: rhythm(0.175),
+      maxWidth: `calc(100% - ${rhythm(0.35)})`,
```
Comment (rowanhogan) [inline source/components/input-search/styles.js]: `right: rhythm(0.175)` 😃

**Comment by danhenderson** [comment]
typeahead2

PR updated @rowanhogan.

I did away with the green box altogether, and just display the selected value similar to a dropdown.

**Comment by danhenderson** [comment]
typeahead3

Updated again 😅

**Comment by rowanhogan** [comment]
I love it!

File: source/components/input-search/index.js [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+
+  /**
+  * If the field is in an error state
+  */
+  error: PropTypes.bool,
+
+  /**
+  * The icon to display on the right of the field (false to hide)
+  */
+  icon: PropTypes.oneOfType([ PropTypes.string, PropTypes.bool ]),
```
Comment (rowanhogan) [inline source/components/input-search/index.js]: 👍

**Pull request merged**